File: frontend/src/entities/category.ts
import { CategoryType } from "./categoryType"
import { Choir } from "./choir"
import { Entity, EntityRef } from "./entity"

export interface Category extends Entity {
    id: number,
    choir: EntityRef<Choir> | undefined,
    name: string,
    categoryType: EntityRef<CategoryType> | undefined,
}

File: backend/src/main/kotlin/nl/stevenbontenbal/chorister/domain/songs/Category.kt
package nl.stevenbontenbal.chorister.domain.songs

import jakarta.persistence.*
import nl.stevenbontenbal.chorister.domain.users.Choir
import nl.stevenbontenbal.chorister.domain.users.ChoirOwnedEntity

@Entity
class Category(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CHOIR_ID")
    override var choir: Choir?,
    var name: String,
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "CATEGORY_TYPE_ID")
    var categoryType: CategoryType
): ChoirOwnedEntity {
    companion object
}